### ISSUE TITLE
RSA-OAEP testing: server provides vectors for subset of hashes

### ISSUE THREAD

**Issue by smuellerDD** [usnistgov/ACVP-Server#71]
This is more a note to NIST than a bug: For RSA-OAEP the server just issues test vectors a subset of the hashes that were registered - we have discussed that before, so that is no news.

But I just found that this can lead to issues: During my development, I only got test vectors for all types of hashes except the registered SHA2-512 and SHA3-512. Now, we obtained vectors that ought to be the final testing and testing fails. Why? It fails because these new vectors now do contain SHA2-512. The reason for the failure is that I requested l=1024 and I requested RSA2048. This, however, is usually not supported because the OAEP padding should be at least twice the hash size. My IUT requires a few bytes more here. So, for SHA2-512 and RSA 2048 and l=1024 this is not guaranteed. So, my IUT fails the vector.

Bottom line: it can be problematic to skip some of the requested ciphers.

**Comment by Kritner** [comment]
Yeah I believe there's a few issues out there related to KAS and only getting a subset of registered capabilities... I'm working on something now that will ensure all capabilities are tested, it's just a bit slow going.  

A full cartesian product of all capabilities for all schemes results in producing files that are too large, but what's currently being produced isn't enough.  I'm trying to find a balance between the two currently.

**Comment by Kritner** [comment]
A change to group level generation around KAS has been pushed to demo for the [v1.1.0.15](https://github.com/usnistgov/ACVP-Server/releases/tag/v1.1.0.15) release.

This change is to use all (rather than a subset) of the registered properties within a registration, without doing a cartesian product of the capabilities.

**Comment by Kritner** [comment]
This change is on prod in release [v1.1.0.15](https://github.com/usnistgov/ACVP-Server/releases/tag/v1.1.0.15)
